### ISSUE TITLE
Autogenerate images in the media center

### ISSUE THREAD

**Issue by barbayellow** [ideascube/ansiblecube#69]
We need to give love to our mediacenter. Very often we forget to add thumbnails for the media in the mediacenter and it is sad. If there no thumbnails we should find a way to add those automatically. There s a script for that : https://wiki.bsf-intranet.org/doku.php?id=ideascube:importmedia#preview

**Comment by fheslouin** [comment]
I think we said that the thumbnails will be generated with the package build with csv2pkg. 
For media mass import we did not choose to generate thumbnails as the set of software to do it was pretty huge...

Do we agree ?

**Comment by bochecha** [comment]
> For media mass import we did not choose to generate thumbnails as the set of software to do it was pretty huge...
> 
> Do we agree ?

This is what I remember as well.

And it's not that we need to install lots of stuff on the servers, it's also that LibreOffice (to generate PDF thumbnails) might be "slightly" too demanding for the Koombooks. 😄 

> I think we said that the thumbnails will be generated with the package build with csv2pkg.

This one I don't remember.

It's certainly possible, however two things to keep in mind:

* the big amount of software to install is still required in this case, just not on the boxes but instead where csv-to-pkg gets run.
* automatically thumbnailing some media could result in poor thumbnails (e.g for movies, which frame to you keep as a thumbnail? the first one? a random one?)

But those are details. I certainly love the result produced by `totem-video-thumbnailer`.

**Comment by fheslouin** [comment]
> This one I don't remember.

I was not clear enough, currently we do this job by hand on a server with all the set of software installed. 

> will be generated with the package build with csv2pkg.

I was meaning we do it in the same time of generating the package. If csv2pkg was able to do that as well that would be even better :)

**Comment by barbayellow** [comment]
> I think we said that the thumbnails will be generated with the package build with csv2pkg.
Don't remember either. However I remember we said the thumbnail will always be in the packages. Generated by BSF on BSF server where we can install anything we want. 

I am really not sure it should be csv2pkg purpose to generate thumbnails since as its name states, csv2pkg is only here to create packages from csv files. Not to generate thumbnails.

**Comment by barbayellow** [comment]
> I think we said that the thumbnails will be generated with the package build with csv2pkg.
Don't remember either. However I remember we said the thumbnail will always be in the packages. Generated by BSF on BSF server where we can install anything we want. 

I am really not sure it should be csv2pkg purpose to generate thumbnails since as its name states, csv2pkg is only here to create packages from csv files. Not to generate thumbnails.

**Comment by mgautierfr** [comment]
Shouldn't be to the omeka plugin (or omeka itself) to generate the thumbnail ?
It would allow better handling of  the thumbnail (changing it if it is not good enough, ...)

**Comment by letompouce** [comment]
When it comes to BSF mass imports, thumbnails are now generated right from the Omeka server (not omeka itself - yet).

Still, a thumbnail should be generated when one uploads some media to the mediacenter.

**Comment by fheslouin** [comment]
Yes that would be nice. The only drawback is the pile of software needed to generate a thumbnail...

**Comment by swbsf** [comment]
Actually I've moved from the totem and evince thumbnailers to other tools with much less dependencies. _ffmpegthumbnailer_ and _convert_. Still quite a few dependencies, but much less ;)

**Comment by bochecha** [comment]
One problem with ffmpeg: patents make it hard to distribute in many parts of the world.

It's something to keep in mind.
